pkg/diagnostics/tracing.go

```diff
@@ -195,9 +195,11 @@ func ConstructSubscriptionSpanAttributes(topic string) map[string]string {
 }
 
 // StartInternalCallbackSpan starts trace span for internal callback such as input bindings and pubsub subscription.
-func StartInternalCallbackSpan(spanName string, parent trace.SpanContext, spec config.TracingSpec) (context.Context, *trace.Span) {
+func StartInternalCallbackSpan(ctx context.Context, spanName string, parent trace.SpanContext, spec config.TracingSpec) (context.Context, *trace.Span) {
 	traceEnabled := diag_utils.IsTracingEnabled(spec.SamplingRate)
-	ctx := context.Background()
+	if ctx == nil {
```

Comment:
I am not sure that we need to set ctx to backgound as a fallback. I would rather setting context to explicitly in caller side.